api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +344,142 @@ type AzureWebhookReceiverConfig struct {
+// and the conditions under which it should be performed.
+type GenericWebhookAction struct {
+	// Name is the name of the action to be performed.
+	//
+	// +kubebuilder:validation:Enum=Refresh;
+	Name GenericWebhookActionName `json:"action" protobuf:"bytes,1,opt,name=action"`
+
+	// MatchExpression is the validation criteria that must be met for the action to
+	// be performed.
+	//
+	// +optional
+	MatchExpression string `json:"matchExpression,omitempty" protobuf:"bytes,2,opt,name=matchExpression"`
+
+	// Parameters contains additional parameters for the action.
+	//
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
+
+	// Targets is a list of selection criteria for the resources on which the
+	// action should be performed.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Targets []GenericWebhookTarget `json:"targets,omitempty" protobuf:"bytes,4,rep,name=targets"`
+}
+
+// GenericWebhookActionName represents the name of an action to be performed on a resource.
+type GenericWebhookActionName string
+
+const (
+	// GenericWebhookActionNameRefresh indicates a request to refresh the resource.
+	GenericWebhookActionNameRefresh GenericWebhookActionName = "Refresh"
+)
+
+// GenericWebhookTarget describes selection criteria for resources to which some
+// action is to be applied.
+type GenericWebhookTarget struct {
+	// Kind is the kind of the target resource.
+	//
+	// +kubebuilder:validation:Enum=Warehouse;
+	Kind GenericWebhookTargetKind `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// Name is the name of the target resource.
+	//
+	// +optional
+	Name string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+
+	// LabelSelector is a label selector to identify the target resources.
+	// If used with IndexSelector, the results are the combined (logical AND) of the two criteria.
+	//
+	// +optional
+	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,3,opt,name=labelSelector"`
+
+	// IndexSelector is a selector used to identify cached target resources by cache key.
+	// If used with LabelSelector, the results are the combined (logical AND) of the two criteria.
+	//
+	// +optional
+	IndexSelector IndexSelector `json:"indexSelector,omitempty" protobuf:"bytes,4,opt,name=indexSelector"`
+}
+
+// GenericWebhookTargetKind represents the kind of a target resource.
+type GenericWebhookTargetKind string
+
+const (
+	GenericWebhookTargetKindWarehouse GenericWebhookTargetKind = "Warehouse"
+)
+
+// IndexSelector encapsulates a selector used to derive index keys
+// based on expressions.
+type IndexSelector struct {
+	// MatchExpressions is a list of index selector requirements.
+	//
+	// +kubebuilder:validation:MinItems=1
+	MatchExpressions []IndexSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,1,rep,name=matchExpressions"`
```

Comment:
Is this what we had in the spec? The field name feels quite wrong, as the selector may actually have nothing at all to do with an expression and simply use static values for all of `key`, `operator`, and `value`.